runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishFactory.java

```diff
@@ -388,6 +429,13 @@ private void onMqttData(
                     .timestamp(now().toEpochMilli())
                     .partition(p -> p.partitionId(-1).partitionOffset(-1))
                     .key(b -> b.set(key))
+                    .hashKey(b ->
+                    {
+                        if (hashKey != null)
+                        {
+                            b.set(hashKey);
+                        }
+                    })
```

Comment:
Create a private method to use instead of inlining here since `hashKey` is a member field.